### PR TITLE
Added cosh tensor operator

### PR DESCRIPTION
**Pull request by Roee-87**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [ x] No

### CODE REVIEW

File: docs/apis/compatibility.md
Comment (franalgaba) [inline docs/apis/compatibility.md:1]: Need to update the % of compatibility

File: src/operators/tensor/core.cairo [hunk 1]
```diff
@@ -1106,10 +1106,78 @@ trait TensorTrait<T> {
     ///     let result = tensor.cosh();
     ///     return result;
     /// }
-    /// >>> [1,0.5403,3.7622]
+    /// >>> [1,1.5403,3.7622]
     /// ```
     ///
     fn cosh(self: @Tensor<T>) -> Tensor<FixedType>;
+    /// #tensor.sinh
```
Comment (franalgaba) [inline src/operators/tensor/core.cairo:1113]: This docstring is a copy paste of `sinh`. Please, update accordingly to `cosh`

File: src/operators/tensor/core.cairo [hunk 2]
```diff
@@ -1106,10 +1106,78 @@ trait TensorTrait<T> {
...
+    ///     // (exp(val)-exp(-val))/2
+    ///     let tensor = fp8x23_tensor_1x3_helper();
+    ///     let result = tensor.sinh();
+    ///     return result;
+    /// }
+    /// >>> [0, 1.1752, 3.6269]
+    /// ```
+    ///
+    fn sinh(self: @Tensor<T>) -> Tensor<FixedType>;
+    /// #tensor.sinh
```
Comment (franalgaba) [inline src/operators/tensor/core.cairo:1147]: Here should be the docstring for `tanh`

File: src/tests/operators/math/cosh/cosh_fp_test.cairo [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    arr.append(FixedTrait::new_unscaled(1, false));
+    arr.append(FixedTrait::new_unscaled(2, false));
+
+    let extra = ExtraParams { fixed_point: Option::Some(FixedImpl::FP8x23(())) };
+
+    let tensor = TensorTrait::<FixedType>::new(sizes.span(), arr.span(), Option::Some(extra));
+
+    let result = tensor.cosh().data;
+
+    // let result1 = *result.at(0);
```
Comment (franalgaba) [inline src/tests/operators/math/cosh/cosh_fp_test.cairo:28]: Why is this commented? If not needed / used better remove the code :)

File: src/tests/operators/math/tanh/tanh_i32_test.cairo [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    data.append(IntegerTrait::new(0_u32, false));
+    data.append(IntegerTrait::new(1_u32, false));
+    data.append(IntegerTrait::new(2_u32, false));
+    let extra = Option::<ExtraParams>::None(());
+
+    let tensor = TensorTrait::<i32>::new(sizes.span(), data.span(), extra);
+
+    let result = tensor.tanh().data; 
+
+    // let result1 = *result.at(0);
```
Comment (franalgaba) [inline src/tests/operators/math/tanh/tanh_i32_test.cairo:30]: Same as other comment

File: src/operators/tensor/core.cairo (hunk 2 above)
Comment (Roee-87) [reply src/operators/tensor/core.cairo:1147]: Ah I just forgot to remove the print() when I was debugging...I removed all print()'s and also removed the import for 'use debug::PrintTrait' in all of the tests that still had it.

File: src/tests/operators/math/cosh/cosh_fp_test.cairo (hunk 3 above)
Comment (Roee-87) [reply src/tests/operators/math/cosh/cosh_fp_test.cairo]: Ah I just forgot to remove the print() when I was debugging...I removed all print()'s and also removed the import for 'use debug::PrintTrait' in all of the tests that still had it.

**Comment by raphaelDkhn** [comment]
Great job, @Roee-87! Your contribution is truly appreciated.

1) I've dropped some thoughts regarding the use of shadowing in the comments section.

2) Could add missing tests to cover all dims by including 1D, 2D, and 3D tensors please? This is an approach we introduced recently (I think after you published this PR) We're aiming to make standard practice moving forward.

3) Also, next time, for ease of review, it would be beneficial if you could split your PRs for each operator implementation.

Thank you once again for your hard work and dedication!

**Comment by raphaelDkhn** [comment]
Thanks for your PR @Roee-87 ! Please do not implement trigonometric ops at the tensor level. We should have other PRs to implement trigo ops in FixedPoint API. 

We should copy paste the code from Cubit lib, and readjust coefficients if needed: 
https://github.com/influenceth/cubit/blob/175d1c9bbf2a929d54759544d9241439c6976f2f/src/math/hyp.cairo#L20

**Pull request closed without merging**